src/optimizer/OptRule.cpp

```diff
@@ -69,10 +71,39 @@ StatusOr<MatchedResult> OptRule::match(const OptGroupNode *groupNode) const {
 }
 
 bool OptRule::match(const MatchedResult &matched) const {
-    UNUSED(matched);
-    // Return true if subclass doesn't override this interface,
-    // so optimizer will only check whether pattern is matched
-    return true;
+    return checkDataflowDeps(matched, matched.node->node()->outputVar(), true);
+}
+
+bool OptRule::checkDataflowDeps(const MatchedResult &matched,
+                                const std::string &var,
+                                bool isRoot) const {
+    auto node = matched.node;
+    auto planNode = node->node();
+    const auto &outVarName = planNode->outputVar();
+    if (outVarName != var) {
```

Comment:
the `outVarName ` is always equal to `var`, they are the same one.

Reply:
Not always, this pr check whether the output variable of dependency is input variable of current plan node in line 101 and line 105.

Reply:
> the `outVarName ` is always equal to `var`, they are the same one.

they are just the same in the first invoke.